File: PracticeManagement/PracticeManagement/Controls/Generic/SelectCutOff/SelectCutOffExtender.cs
using System.Web.UI;
using AjaxControlToolkit;

[assembly: WebResource("PraticeManagement.Controls.Generic.SelectCutOff.SelectCutOffExtender.js", "text/javascript")]

namespace PraticeManagement.Controls.Generic.SelectCutOff
{
    using System.ComponentModel;

    /// <summary>
    /// Extender that fixes width of select element in IE.
    /// </summary>
    [ClientScriptResource("PraticeManagement.Controls.Generic.SelectCutOff.SelectCutOffBehavior", "PraticeManagement.Controls.Generic.SelectCutOff.SelectCutOffExtender.js")]
    [TargetControlType(typeof(Control))]
    public class SelectCutOffExtender : ExtenderControlBase
    {
        [ExtenderControlProperty]
        [DefaultValue("")]
        [ClientPropertyName("normalCssClass")]
        public string NormalCssClass
        {
            get { return GetPropertyValue("normalCssClass", string.Empty); }
            set { SetPropertyValue("normalCssClass", value); }
        }

        [ExtenderControlProperty]
        [DefaultValue("")]
        [ClientPropertyName("extendedCssClass")]
        public string ExtendedCssClass
        {
            get { return GetPropertyValue("extendedCssClass", string.Empty); }
            set { SetPropertyValue("extendedCssClass", value); }
        }
    }
}

File: PracticeManagement/PracticeManagement/Controls/Generic/SelectCutOff/SelectCutOffExtender.js
Type.registerNamespace("PraticeManagement.Controls.Generic.SelectCutOff");

PraticeManagement.Controls.Generic.SelectCutOff.SelectCutOffBehavior = function(element) {
    PraticeManagement.Controls.Generic.SelectCutOff.SelectCutOffBehavior.initializeBase(this, [element]);
}

PraticeManagement.Controls.Generic.SelectCutOff.SelectCutOffBehavior.prototype = {
    initialize: function() {
        PraticeManagement.Controls.Generic.SelectCutOff.SelectCutOffBehavior.callBaseMethod(this, 'initialize');

        $addHandlers(this.get_element(),
                     { 'mousedown': this._onFocus,
                         'blur': this._onBlur,
                         'change': this._onBlur
                     },
                     this);
    },

    dispose: function() {
        // Cleanup code 
        PraticeManagement.Controls.Generic.SelectCutOff.SelectCutOffBehavior.callBaseMethod(this, 'dispose');
    },

    _onFocus: function() {
        var target = this.get_element();
        if (target != null) {
            target.setAttribute("class", this.extendedCssClass);
        }
    },

    _onBlur: function() {
        var target = this.get_element();
        if (target != null) {
            target.setAttribute("class", this.normalCssClass);
        }
    }
}

PraticeManagement.Controls.Generic.SelectCutOff.SelectCutOffBehavior.registerClass('PraticeManagement.Controls.Generic.SelectCutOff.SelectCutOffBehavior', AjaxControlToolkit.BehaviorBase);
